### PR TITLE
K8SPSMDB-257: introduce cr version field

### PR DESCRIPTION
**Pull request by qJkee**
[K8SPSMDB-257](https://jira.percona.com/browse/K8SPSMDB-257)

Add crVersion field which shows current version of psmdb cluster

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade | passed |
| self-healing | passed |
| monitoring | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| scheduled-backup | passed |
| arbiter | failed |
| service-per-pod | passed |
| liveness | passed |
| users | passed |

**Review by delgod — DISMISSED** [review]

File: pkg/apis/psmdb/v1/psmdb_types.go [hunk 1]
```diff
@@ -469,41 +468,37 @@ func (cr *PerconaServerMongoDB) OwnerRef(scheme *runtime.Scheme) (metav1.OwnerRe
 // The new (semver-matching) version is determined either by the CR's API version or an API version specified via the CR's annotations.
 // If the CR's API version is an empty string, it returns "v1"
 func (cr *PerconaServerMongoDB) setVersion() error {
-	apiVersion := cr.APIVersion
+	if len(cr.Spec.CRVersion) > 0 {
+		return nil
+	}
```
Comment (blez) [inline pkg/apis/psmdb/v1/psmdb_types.go:473]: What if user will want to apply configuration with new CR version?

P.S. I'm just interested, not sure it's valid case.

Comment (qJkee) [reply pkg/apis/psmdb/v1/psmdb_types.go:473]: Ahh, i forget to add example to the cr.yaml =)
But basically, you can change this value from yaml, according to the update docs

**Review by blez — APPROVED** [review]

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-431-c02a57a2

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| operator-self-healing | passed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| one-pod | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| service-per-pod | passed |
| liveness | passed |
| users | passed |

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-431-b9498ddf

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade | passed |
| self-healing | passed |
| monitoring | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| arbiter | passed |
| scheduled-backup | passed |
| service-per-pod | passed |
| liveness | passed |
| users | passed |

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-431-131ba61f

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade | passed |
| self-healing | passed |
| monitoring | passed |
| upgrade-consistency | passed |
| monitoring-2-0 | passed |
| limits | passed |
| demand-backup | passed |
| scaling | passed |
| operator-self-healing | passed |
| scheduled-backup | passed |
| security-context | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| users | passed |

**Review by blez — APPROVED** [review]

**Review by delgod — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade | passed |
| self-healing | passed |
| monitoring | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| arbiter | passed |
| service-per-pod | passed |
| scheduled-backup | passed |
| liveness | passed |
| users | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring | passed |
| storage | passed |
| monitoring-2-0 | passed |
| self-healing | passed |
| arbiter | passed |
| operator-self-healing | passed |
| service-per-pod | passed |
| liveness | passed |
| users | failed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| monitoring | passed |
| self-healing | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| users | failed |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |

**Review by blez — DISMISSED** [review]

**Review by hors — CHANGES_REQUESTED** [review]
I think we need to set 
```
name: v1-6-0
storage: true
```
for PSMDBO too

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade | passed |
| self-healing | passed |
| monitoring | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| security-context | passed |
| arbiter | passed |
| scheduled-backup | passed |
| service-per-pod | passed |
| liveness | passed |
| users | failed |

**Review by delgod — APPROVED** [review]

**Pull request merged**